aws/delete_s3_velero_bucket.py

```diff
@@ -0,0 +1,234 @@
+
+    Args:
+        bucket_name: The name of the cluster bucket to delete
+
+    Returns:
+        velero infrastructure name or None
+    """
+
+    LOGGER.info(f"Get tags for bucket: {bucket_name}")
+    bucket_tags = S3_CLIENT.get_bucket_tagging(Bucket=bucket_name)["TagSet"]
+
+    for tag in bucket_tags:
+        if tag["Key"] == "velero.io/infrastructureName":
+            return tag["Value"]
+
+
+def verify_cluster_matches_velero_infrastructure_name(
+    cluster_name: str,
+    bucket_name: str,
+) -> bool:
+    """
+    Get the velero bucket infrastructure name and compare it with the cluster
+
+    Args:
+        cluster_name: The cluster name
+        bucket_name: The bucket name
+
+    Returns:
+        True if the cluster matches a velero infrastructure name, False otherwise
+    """
+
+    LOGGER.info(
+        f"Verify cluster matches a velero infrastructure name via its bucket tag: {bucket_name}",
+    )
+
+    velero_infrastructure_name = get_velero_infrastructure_name(bucket_name=bucket_name)
+
+    # Verify if the bucket is associated with the cluster
+    if velero_infrastructure_name and re.search(
+        rf"{cluster_name}(-\w+)?$",
+        velero_infrastructure_name,
+    ):
+        LOGGER.info(
+            f"Verified cluster '{cluster_name}' "
+            f"is associated with velero infrastructure name {velero_infrastructure_name}",
+        )
+        return True
+
+    return False
+
+
+def delete_all_objects_from_s3_folder(bucket_name: str) -> None:
+    """
+    Deletes all files in a folder of an S3 bucket
+
+    Args:
+        bucket_name: The bucket name
+    """
+
+    list_obj_response = S3_CLIENT.list_objects_v2(Bucket=bucket_name)
+
+    #  Sometimes there maybe no contents -- no objects
+    files_in_folder = list_obj_response.get("Contents")
+    if not files_in_folder:
+        LOGGER.info(f"No objects to be deleted for {bucket_name}")
+        return
+
+    files_to_delete = [{"Key": file["Key"]} for file in files_in_folder]
+    delete_response = S3_CLIENT.delete_objects(
+        Bucket=bucket_name,
+        Delete={"Objects": files_to_delete, "Quiet": True},
+    )
+
+    delete_response_http_status_code = delete_response["ResponseMetadata"][
+        "HTTPStatusCode"
+    ]
+    if delete_response_http_status_code == HTTPStatus.OK:
+        LOGGER.info("Objects deleted successfully")
+        return
+    else:
+        LOGGER.error(
+            "Objects not deleted:\n:"
+            f"{json.dumps(delete_response, default=str, indent=4)}",
+        )
+        sys.exit(1)
+
+
+def main():
+    found_env = []
+    for env_var in REQUIRE_ENV:
+        if os.getenv(env_var):
+            found_env.append(env_var)
+    if len(found_env) < len(REQUIRE_ENV):
+        LOGGER.error(
+            f"Missing environment variable found: {found_env}, require:{REQUIRE_ENV}"
+        )
+        sys.exit(1)
+
+    parser = argparse.ArgumentParser(description="Delete an s3 velero cluster bucket")
```

Comment:
I think you can set args as required and set default to them
then you do not need to check the REQUIRE_ENV, set arg for each as required and have default for each

also if `--cluster_name` is also required please set it as required

Reply:
using click and all args are required, AWS parameters are set as env variable and set as default.